cmake/Findclang.cmake

```diff
@@ -5,11 +5,12 @@
 # CLANG_FOUND
 # CLANG_INCLUDE_DIRS
 # CLANG_LIBRARIES
-# CLANG_LIBDIRS
 
-if(MSVC)
-  find_package(CLANG REQUIRED CONFIG)
+find_package(Clang REQUIRED)
 
+if(CLANG_LINK_CLANG_DYLIB)
```

Comment:
I don't think this option is supported by clang < 9. Perhaps target this PR at the zig llvm9 branch?

Reply:
No. Users have to specify `-DCLANG_LINK_CLANG_DYLIB` explicitly to enable this code path. The code is here so that:

* after the global constructor conflict caused by the embedded lld is fixed
* after the upstream exports `CLANG_LINK_CLANG_DYLIB` as cmake variable

This can work automatically.